### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by rroohhh**

### CODE REVIEW

**Review by anuejn — APPROVED** [review]
looks good :)

**Comment by rroohhh** [comment]
This now switches to the fastapi security stuff, which is also integrated with openapi, but `openapi-typescript-fetch` doesn't support that I think.

So maybe we want to build support for this sometime.

**Comment by anuejn** [comment]
hm... lets not use it then? sounds a bit cumbersome for not a lot of gain

**Comment by rroohhh** [comment]
Maybe, but apart from the `openapi-typescript-fetch` thing this is a lot more proper.

File: backend/transcribee_backend/auth.py [hunk 1]
```diff
@@ -66,11 +65,19 @@ def validate_user_authorization(session: Session, authorization: str):
...
-):
-    return validate_user_authorization(session, authorization)
+class TokenAuth(HTTPBase):
+    def __init__(self):
+        self.model = HTTPBaseModel(scheme="token")
+        self.scheme_name = "Token"
+
+    async def __call__(
+        self, request: Request, session: Session = Depends(get_session)
+    ) -> Optional[UserToken]:
```
Comment (pajowu) [inline backend/transcribee_backend/auth.py:75]: Is this really `Optional`? I think we always return a `UserToken`, otherwise we raise an Exception

**Comment by pajowu** [comment]
I'm not sure about fastapi.security as well, as long as our frontend (the only place we really need this) doesn't work with it? And if we switch, should we switch the worker auth as well?

**Comment by pajowu** [comment]
The other stuff looks good to me

**Comment by rroohhh** [comment]
I dropped the fastapi.security stuff and moved it to #140

File: backend/transcribee_backend/auth.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ def validate_user_authorization(session: Session, authorization: str):
         raise HTTPException(status_code=401)
 
     try:
-        token_data = b64decode(token.encode()).decode()
```
Comment (pajowu) [inline backend/transcribee_backend/auth.py:53]: Why this?

Comment (rroohhh) [reply backend/transcribee_backend/auth.py:53]: `b64decode` already internally encodes ascii strings, so this seems unnecessary?

**Pull request merged**